src/UCTSearch.cpp

```diff
@@ -235,6 +235,39 @@ void UCTSearch::dump_stats(FastState & state, UCTNode & parent) {
     tree_stats(parent);
 }
 
+void UCTSearch::output_analysis(FastState & state, UCTNode & parent) {
+    if (!parent.has_children()) {
+        return;
+    }
+
+    const int color = state.get_to_move();
+    
+    // sort children, put best move on top
+    parent.sort_children(color);
+    
+    if (parent.get_first_child()->first_visit()) {
+        return;
+    }
+    
+    int movecount = 0;
+    std::string separator = "info ";
+    for (const auto& node : parent.get_children()) {
+        // Always display at least two moves. In the case there is
+        // only one move searched the user could get an idea why.
+        if (++movecount > 2 && !node->get_visits()) break;
+        
+        std::string move = state.move_to_text(node->get_move());
+        FastState tmpstate = state;
+        tmpstate.play_move(node->get_move());
+        std::string pv = move + " " + get_pv(tmpstate, *node);
+        gtp_printf_raw("%s%s %d %.2f %s", separator.c_str(), move.c_str(),node->get_visits(),
```

Comment:
I would really put tags in here like UCI does, i.e. "info move d4 visits 100 winrate 5000" instead of relying on the position or order. Else it gets really brittle if there's ever a need to display more info about a move, or if we don't know some of the info. With tags, the GUI can just parse and display those it understands.

(percentage in permilles)

Reply:
I like that. info move d4 visits 100 winrate 5000 variation d4 d6 d2 c2
Now I've grouped all the alternatives on one line using the | as separator. Should we just use start again with the info tag, or just the move tag, or a different separator all together?

info move d4 visits 100 winrate 5000 pv d4 d6 d2 c2 info move d2 visits 60 winrate 4900 pv d3 d7 e3 e4
or just 
info move d4 visits 100 winrate 5000 pv d4 d6 d2 c2 move d2 visits 60 winrate 4900 pv d3 d7 e3 e4
(without the extra info)?
(What is pv short for btw? Play variation?)

Reply:
PV = Principal Variation. (This is the standard term and abbreviation in computer chess as well.)